btcutil/hdkeychain/extendedkey.go

```diff
@@ -292,8 +291,8 @@ func (k *ExtendedKey) Derive(i uint32) (*ExtendedKey, error) {
 	// chance (< 1 in 2^127) this condition will not hold, and in that case,
 	// a child extended key can't be created for this index and the caller
 	// should simply increment to the next index.
-	ilNum := new(big.Int).SetBytes(il)
-	if ilNum.Cmp(btcec.S256().N) >= 0 || ilNum.Sign() == 0 {
+	var ilNum btcec.ModNScalar
+	if overflow := ilNum.SetByteSlice(il); overflow {
```

Comment:
Note that this is kinda optional, big.Ints work here, but wanted to make things more uniform so started to use the new `ModNScalar`.